src/service/gif.js

```diff
@@ -1,9 +1,13 @@
 import api from './config';
 
-export const getGifFeatured = async () =>
+export const getGifFeatured = async (search) =>
   new Promise((resolve, reject) => {
     api
-      .get('/chat/gif/featured')
+      .get(
+        search?.trim() !== ''
+          ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
+          : '/chat/gif/featured'
+      )
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `getGifFeatured` function is a good improvement as it now supports searching for GIFs. However, there's a potential issue with the way the search query is being encoded and appended to the URL. If the `search` parameter contains special characters, it could lead to unexpected results or even security vulnerabilities such as Injection attacks.

To mitigate this, you should use a more secure method to pass parameters in your API call. Instead of manually constructing the URL, consider passing the parameters as an object to the `params` property in the config object of the `get` method. This way, axios will automatically encode the parameters correctly.

Here's how you can do it:

```diff
- .get(
-   search?.trim() !== ''
-     ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}`
-     : '/chat/gif/featured'
- )
+ .get(
+   search?.trim() !== ''
+     ? '/chat/gif/search'
+     : '/chat/gif/featured',
+   {
+     params: {
+       q: search?.trim()
+     }
+   }
+ )
```
This change ensures that the `search` parameter is properly encoded and mitigates potential security risks.


<!-- This is an auto-generated comment by OSS CodeRabbit -->